cardinal/evm/server.go

```diff
@@ -170,9 +173,16 @@ func (s *msgServerImpl) Serve() error {
 			log.Fatal(err)
 		}
 	}()
+	s.shutdown = server.GracefulStop
 	return nil
 }
 
+func (s *msgServerImpl) Shutdown() {
+	if s.shutdown != nil {
+		s.shutdown()
+	}
+}
+
```

Comment:
perhaps it would be useful to do an `else { panic("evm.Server was shutdown without calling Serve() }` or something of that nature? this feels a tad leaky

Reply:
Per our conversation, I'm treating this "Shutdown" as "make sure the server is not running", so it shouldn't produce an error. Similarly, calling Shutdown multiple times shouldn't produce an error. Go's http.Server.Shutdown follows a similarly pattern.